src/gui/tools/gt_qmltoolbar.cpp

```diff
@@ -0,0 +1,120 @@
+/* GTlab - Gas Turbine laboratory
+ *
+ * SPDX-License-Identifier: MPL-2.0+
+ * SPDX-FileCopyrightText: 2023 German Aerospace Center (DLR)
+ */
+
+#include "gt_qmltoolbar.h"
+
+#include <QDebug>
+#include <QQuickItem>
+
+#include <QLayout>
+#include <QQuickWidget>
+#include <QQmlContext>
+
+#include <iterator>
+
+struct GtQmlToolbar::Impl
+{
+    QVariantListModel m_toolbarGroupsList;
+    GtQmlToolbarGroup m_statusActions;
+    QQuickWidget qmlToolbar;
+
+    bool m_darkMode = {false};
+};
+
+GtQmlToolbar::GtQmlToolbar(QWidget* parent) :
+    QWidget(parent), pimpl(std::make_unique<Impl>())
+{
+    static auto typeRegistered = []() {
+        qmlRegisterType<GtQmlAction>("de.gtlab", 2, 1, "ToolbarAction");
+        return true;
+    }();
+    Q_UNUSED(typeRegistered);
+
+
+    pimpl->qmlToolbar.rootContext()->setContextProperty("toolbar", this);
+    pimpl->qmlToolbar.setSource(QUrl(QStringLiteral("qrc:/qml/Toolbar.qml")));
+
+    auto layout = new QVBoxLayout();
+    layout->addWidget(&pimpl->qmlToolbar);
+    layout->setContentsMargins(0, 0, 0, 0);
+    setLayout(layout);
+
+    pimpl->qmlToolbar.setResizeMode(QQuickWidget::SizeRootObjectToView);
+}
+
+GtQmlToolbar::~GtQmlToolbar() = default;
+
+QVariantListModel*
+GtQmlToolbar::toolbarGroups()
+{
+    return &pimpl->m_toolbarGroupsList;
+}
+
+void
+GtQmlToolbar::addToolbarGroup(GtQmlToolbarGroup* group)
+{
+
+    if (!group)
+        return;
```

Comment:
I think this is our coding convection for such if-statements
```suggestion
    if (!group) return;
```